sdk/storage/azure-storage-file-datalake/dev_requirements.txt

```diff
@@ -0,0 +1,7 @@
+-e ../../../tools/azure-devtools
+-e ../../../tools/azure-sdk-tools
+../../core/azure-core
+../azure-storage-blob
+-e ../../identity/azure-identity
```

Comment:
is this really required?

Reply:
I think we want this to test oauth authentication(eg. we want to import ClientSecretCredential from azure-identity) not sure why blob and queue have identity while fhileshare doesn't have this 
https://github.com/Azure/azure-sdk-for-python/blob/8380bfd325f500aaa2999ac2f96be9f85329bc39/sdk/storage/azure-storage-file-share/dev_requirements.txt#L1-L5

Reply:
This file is a dev file, if you have tests that uses identity, then it needs to be here. This file doesn't impact runtime dependency of the setup.py